### DISCUSSION TITLE
Keycloak on windows using kc.bat returns error "unexpected at this..."

### DISCUSSION THREAD

**Discussion by AmsalkhirAli** [keycloak/keycloak#19219] [Q&A]
I downloaded the Keycloak zip and followed the instructions in the documentation to run in windows. When running in a folder without spaces keycloak starts without any issue. but when putting the keycloak folder in c:\program files(x86)\ I can not start keycloak anymore using kc.bat
I get the following error
**"JAVA_OPTS already set in environment; overriding default settings with values: -Dprogram.name=kc.bat -Dprogram.name=kc.bat -Dprogram.name=kc.bat -Dprogram.name=kc.bat -Xms64m -Xmx512m -XX:MetaspaceSize=96M -XX:MaxMetaspaceSize=256m -Djava.net.preferIPv4Stack=true -Dfile.encoding=UTF-8"
\PID\keycloak-20.0.3\keycloak-20.0.3\bin\.." -Djboss.server.config.dir="C:\Program was unexpected at this time.** 

Someone knows how to fix this please?

**Comment by ssilvert** [comment]
I just tried a path with a space on my Windows machine using a recent nightly build and it worked fine.

Maybe try the nightly?
https://github.com/keycloak/keycloak/releases/tag/nightly

**Reply by AmsalkhirAli** [reply]
Cool thanks for the quick response. I tested the nightly build and this seems to work under c:\Program files\ folder but not under the c:\program files (x86)\
C:\Program Files (x86)\keycloak-999.0.0-SNAPSHOT\bin>kc.bat start-dev
\keycloak-999.0.0-SNAPSHOT\bin\.." -Djboss.server.config.dir="C:\Program was unexpected at this time.

**Reply by sschu** [reply]
You could give my branch in this PR a try (just use the kc.bat from there: https://github.com/keycloak/keycloak/pull/16229).

**Reply by AmsalkhirAli** [reply]
Hello, Thanks for the help. It is working with spaces. But doing a bit more trials, I found out that the culprit is the closing round bracket character `)`. So starting Keycloak using kc.bat in a folder named `c:\Program Files(x86\ `works perfectly but once I close the bracket and run it from a folder called `c:\Program Files(x86)\` I get an error. c:\Program unexpected at this time

**Comment by ssilvert** [comment]
@AmsalkhirAli Can you open a [bug report](https://github.com/keycloak/keycloak/issues/new/choose) for this?

**Reply by AmsalkhirAli** [reply]
yes it's done
